File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/cpp_solutions/2935-maximum-strong-pair-xor-ii.cpp
class Solution {
public:
    int maximumStrongPairXor(vector<int>& nums) {
        int max_xor = 0;
        int n = nums.size();
        for (int i = 0; i < n; ++i) {
            for (int j = i; j < n; ++j) {
                if (abs(nums[i] - nums[j]) <= min(nums[i], nums[j])) {
                    max_xor = max(max_xor, nums[i] ^ nums[j]);
                }
            }
        }
        return max_xor;
    }
};

File: leetcode/AIcoding/2.Gemini-2.0/javascript_solutions/2935-maximum-strong-pair-xor-ii.js
var maximumStrongPairXor = function(nums) {
    nums.sort((a, b) => a - b);
    let max_xor = 0;
    let j = 0;
    for (let i = 0; i < nums.length; i++) {
        while (j < nums.length && nums[j] <= 2 * nums[i]) {
            j++;
        }
        for (let k = i; k < j; k++) {
            if (Math.abs(nums[i] - nums[k]) <= Math.min(nums[i], nums[k])) {
                max_xor = Math.max(max_xor, nums[i] ^ nums[k]);
            }
        }
    }
    return max_xor;
};